src/k8s-engine/wsl.ts

```diff
@@ -1233,6 +1238,7 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
             await this.execCommand('mkdir', '-p', '/var/lib/misc');
 
             await this.runInit();
+            lock.kill('SIGTERM');
```

Comment:
This should always run, so there should be a `try/finally` somewhere.

I thinking something like
```ts
try {
  await Promise.all(installerActions);
} finally {
  lock.kill('SIGTERM');
}
```
Note that this code is Windows-only, so [according to docs](https://nodejs.org/docs/latest-v16.x/api/child_process.html#subprocesskillsignal) the arugment here doesn't matter and the process is always killed with a `TerminateProcess`.